### PR TITLE
Use puppeteer and update dependencies

### PR DESCRIPTION
**Pull request by AlejandroPerezMartin**

### CODE REVIEW

File: routes/main.js [hunk 1]
```diff
@@ -12,38 +13,68 @@ exports.main = {
...
 
-    if (request.query.viewport) {
-      command.push(`--window-size=${request.query.viewport}`);
-    }
+      if (request.query.viewport) {
+        const [width, height] = request.query.viewport.split(',');
+        
+        await page.setViewport({
+          width: parseInt(width, 10) || 1200,
+          height: parseInt(height, 10) || 800,
```
Comment (Antonio-Laguna) [inline routes/main.js]: This should be `900` as per docs.

File: routes/main.js [hunk 2]
```diff
@@ -12,38 +13,68 @@ exports.main = {
...
+      await page.screenshot({
+        path: outputFile,
+        fullPage: true
+      });
+
+      await browser.close();
+
+      server.log(['debug'], outputFile);
+
+      Promise.resolve(outputFile);
```
Comment (Antonio-Laguna) [inline routes/main.js]: What is this?

File: routes/main.js [hunk 3]
```diff
@@ -12,38 +13,68 @@ exports.main = {
...
+
+      return reply.file(outputFile);
+    }
+    catch (e) {
+        console.log(`Error while parsing "${request.query.url}"`, e);
+        server.log(['error'], e);
+
+        return Promise.resolve(null);
```
Comment (Antonio-Laguna) [inline routes/main.js]: No need for this either. Just return `null`

File: routes/main.js [hunk 4]
```diff
@@ -12,38 +13,65 @@ exports.main = {
...
     const url = request.query.url;
     const randFldr = rando(18);
     const dir = `./screenshots/${randFldr}`;
+    const outputFile = `${dir}/screenshot.png`;
+    
+    try {
+      let browser;
 
-    const command = [];
+      fs.mkdirSync(dir);
```
Comment (Antonio-Laguna) [inline routes/main.js]: Just seen this, can we use `fs.mkdir` use `util.promisify` to make it `await`able.

File: routes/main.js [hunk 5]
```diff
@@ -12,38 +13,65 @@ exports.main = {
...
-    const command = [];
+      await fs.ensureDir(dir);
+      
+      if (request.query.wsEndpoint) {
+        console.log(`Connecting to browser websocket at "${request.query.wsEndpoint}".`);
+        browser = await puppeteer.connect({
+          browserWSEndpoint: request.query.wsEndpoint
+        });
+      } else {
+        browser = await puppeteer.launch();
```
Comment (ecwillis) [inline routes/main.js]: Should add and ENV Var as a default so we don't have to pass the `wsEndpoint` to the app every time.

**Review by ecwillis — CHANGES_REQUESTED** [review]

**Pull request merged**